File: leetcode/1784-check-if-binary-string-has-at-most-one-segment-of-ones/checkOnesSegment.ts
function checkOnesSegment(s: string): boolean {
  let recordIndex = -2;
  let onesCount = 0;

  for (const [index, numString] of Array.from(s).entries()) {
    if (numString != "1") {
      continue;
    }

    if (recordIndex + 1 == index) {
      recordIndex = index;
      continue;
    }

    recordIndex = index;
    onesCount++;
  }

  return onesCount <= 1;
}

export default checkOnesSegment;

File: leetcode/1784-check-if-binary-string-has-at-most-one-segment-of-ones/checkOnesSegment.go
package leetcode

func checkOnesSegment(s string) bool {
	var onesCount = 0
	var recordIndex = -2

	for index, numString := range s {
		if numString != '1' {
			continue
		}

		if recordIndex+1 == index {
			recordIndex = index
			continue
		}

		recordIndex = index
		onesCount++
	}

	return onesCount <= 1
}
